### PR TITLE
Refactor: Retain filter #1024

### LINKED ISSUE #1024: Feat: Maintaining the filter
### What feature?

Currently, the projects are filtered if we apply any of the filters.
If we click the `More ->` button on any of the filtered projects, we will be able to see all the projects of the user.
On this page, there will be a `< Back to Projects` button if we click on that button we'll be taken back to the projects page.
But, the filter we applied earlier will be gone. 
We can improve this and maintain the applied filter when the user clicks on the `< Back to Projects` button.

### Add screenshots

No screenshots.

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by kailashchoudhary11**
fixes #1024

### CODE REVIEW

**Comment by kailashchoudhary11** [comment]
@priyankarpal level2 lag rha hai ye code kam hai but kaafi tricky hai 🤔

File: src/components/ProjectList.jsx [hunk 1]
```diff
@@ -40,8 +42,8 @@ const ProjectList = () => {
           {/* Beack to projects link */}
           <div className="m-4 hover:text-purple-500 transition-all duration-300 ease-in-out">
             <span>{'<'}</span>
-            <Link to="/projects" className="font-mono ml-2">
-              Back to Projects
+            <Link to={`/projects${filter ? `?filter=${filter}` : ''}`} className="font-mono ml-2">
+              {`Back to ${filter ? filter.charAt(0).toUpperCase() + filter.slice(1) : 'All'} Projects`}
```
Comment (priyankarpal) [inline src/components/ProjectList.jsx:46]: remove the `mono font`  we are using `inter` by default

``` suggestion
<Link to={`/projects${filter?`?filter=${filter}`:''}`} className="ml-2">{`Back to ${filter?filter.charAt(0).toUpperCase()+filter.slice(1):'All'} Projects`}</Link>

```

**Review by priyankarpal — CHANGES_REQUESTED** [review]

**Comment by kailashchoudhary11** [comment]
Done @priyankarpal

**Review by priyankarpal — APPROVED** [review]
Looks Good 🎉

**Review by priyankarpal — APPROVED** [review]

**Pull request merged**